### DISCUSSION TITLE
"The change you are trying to make would violate the required relation" when trying to delete parent in explicit M2M, but able to delete with raw SQL

### DISCUSSION THREAD

**Discussion by hi019** [prisma/prisma#2610] [Q&A] [answered]
**Minimal repo with reproducible issue: https://github.com/hi019/prisma-issue**

Hello, I am having an issue with a one-to-many relation (well, explicit many-to-many). Here is my schema: 

```
model Event {
  id                  Int                @default(autoincrement()) @id
  name                String?
  ownerId             Int
  owner               User               @relation(fields: [ownerId], references: [id])
  users         EventUser[]
}

model EventUser {
  eventId Int
  userId  Int
  event   Event @relation(fields: [eventId], references: [id])
  user    User  @relation(fields: [userId], references: [id])

  @@id([userId, eventId])
}

model User {
  id    Int     @id @default(autoincrement())
  email String  
}

```

I insert a `User`, then an `Event` which belongs to that user, and finally creating an `EventUser` for the event and user. Finally I try to delete the event, but get `The change you are trying to make would violate the required relation 'EventToEventUser' between the 'Event' and 'EventUser' models.`. Here is the code, without the boilerplate async function and catch [[full code](https://github.com/hi019/prisma-issue/blob/master/script.ts)]:

```ts
  await prisma.connect()
  
  const {id: userId} = await prisma.user.create({
    data: {
      email: 'test@test.com',
    }
  })

  const { id: eventId } = await prisma.event.create({
    data: {
        owner: { connect: { id: userId } },
        name: 'test',
    },
  });

  await prisma.eventUser.create({
    data: {
        event: { connect: { id: eventId } },
        user: { connect: { id: userId } },
    },
});

  await prisma.event.delete({
    where: {id: eventId}
  })
```

When an `Event` is deleted, I would like all `EventUser`s that reference the deleted `Event` be removed as well. From my limited SQL knowledge, it seems that is what's supposed to be happening according to the migration (CASCADE) Prisma created, [here](https://github.com/hi019/prisma-issue/tree/master/prisma/migrations/20200529214610-initial)

Also, in the mysql console, doing `DELETE FROM Event WHERE id=3` (with id=3 being the event that was created in the script above) successfully deletes the event and the eventUsers referencing it. Maybe this problem is related to [this](https://github.com/prisma/prisma/issues/2057) issue?

**Comment by ryands17** [comment, marked as answer]
Hey @hi019 :wave: 
Yes this is an open issue for now, `prisma.raw` is the workaround that would have to be taken until this is solved.

**Reply by hi019** [reply]
Got it. For anyone wondering, the Github Issue for this is #2328. It's currently assigned to the Beta 8 milestone.

**Comment by ludralph** [comment]
Hi @hi019,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
